src/client.ts

```diff
@@ -405,42 +428,93 @@ export class UXDClient {
     const mangoPerpMarketConfig = mango.getPerpMarketConfig(
       depository.collateralMintSymbol
     );
-    const userCollateralATA = findATAAddrSync(
-      user,
-      depository.collateralMint
-    )[0];
-    // Redeemable ATA
+    const redeemAmountBN = uiToNative(
+      amountRedeemable,
+      controller.redeemableMintDecimals
+    );
+
+    return this.instruction.redeemFromMangoDepository(
+      redeemAmountBN,
+      limitPriceI80F48.toNumber(),
+      {
+        accounts: {
+          user,
+          payer: payer ?? user,
+          controller: controller.pda,
+          depository: depository.pda,
+          collateralMint: depository.collateralMint,
+          redeemableMint: controller.redeemableMintPda,
+          userCollateral: userCollateralATA,
+          userRedeemable: userRedeemableATA,
+          mangoAccount: depository.mangoAccountPda,
+          // mango stuff
+          mangoGroup: mango.group.publicKey,
+          mangoCache: mangoCacheAccount,
+          mangoSigner: mangoGroupSigner,
+          // -- for the withdraw
+          mangoRootBank: mangoRootBankAccount,
+          mangoNodeBank: mangoNodeBankAccount,
+          mangoVault: mangoDepositedVaultAccount,
+          // -- for the perp position closing
+          mangoPerpMarket: mangoPerpMarketConfig.publicKey,
+          mangoBids: mangoPerpMarketConfig.bidsKey,
+          mangoAsks: mangoPerpMarketConfig.asksKey,
+          mangoEventQueue: mangoPerpMarketConfig.eventsKey,
+          //
+          systemProgram: SystemProgram.programId,
+          tokenProgram: TOKEN_PROGRAM_ID,
+          mangoProgram: mango.programId,
+        },
+        options: options,
+      }
+    );
+  }
+
+  public async createQuoteMintWithMangoDepositoryInstruction(
+    quoteAmount: number,
+    controller: Controller,
+    depository: MangoDepository,
+    mango: Mango,
+    user: PublicKey,
+    options: ConfirmOptions,
+    payer?: PublicKey
+  ): Promise<TransactionInstruction> {
+    const depositedTokenIndex = mango.group.getTokenIndex(depository.quoteMint);
+    const mangoCacheAccount = mango.getMangoCacheAccount();
+    const mangoRootBankAccount = mango.getRootBankForToken(depositedTokenIndex);
+    const mangoNodeBankAccount = mango.getNodeBankFor(
+      depositedTokenIndex,
+      depository.quoteMint
+    );
+    const mangoDepositedVaultAccount = mango.getVaultFor(depositedTokenIndex);
+    const mangoPerpMarketConfig = mango.getPerpMarketConfig(
+      depository.collateralMintSymbol
+    );
+    const userQuoteATA = findATAAddrSync(user, depository.quoteMint)[0];
```

Comment:
`findATAAddrSync` or `utils.token.associatedAddress` ?

There can be only one.

Reply:
FindAtaAddrSync let's not change what doesn't need to be change

Reply:
https://github.com/UXDProtocol/uxd-client/pull/3#discussion_r875697863 if that's ok